framework/src/media/audio/audio_manager.c

```diff
@@ -123,15 +129,37 @@ audio_manager_result_t find_input_audio_card(void)
 			|| (dir_entry->d_name[1] != 'c')
 			|| (dir_entry->d_name[2] != 'm')
 			|| (sscanf(&dir_entry->d_name[3], "C%uD%u%c", &card_id, &device_id, &type) != 3)) {
-			closedir(dir_info);
-			return AUDIO_MANAGER_INVALID_DEVICE_NAME;
+			continue;
 		}
 
 		if ((type == 'c') && (card_id < CONFIG_AUDIO_MAX_INPUT_CARD_NUM) && (device_id < MAX_AUDIO_DEVICE_NUM)) {
+#ifndef CONFIG_AUDIO_EXCLUDE_VOLUME
+			struct audio_caps_desc_s caps_desc;
+			FAR struct audio_caps_s *caps = &(caps_desc.caps);
+			int fd = open(strcat(path, dir_entry->d_name), O_RDONLY);
+
+			if (fd < 0) {
+				auddbg("Fail to open input driver, path = %s\n", path);
+				return AUDIO_MANAGER_FAIL;
+			}
+
+			caps->ac_len = sizeof(struct audio_caps_s);
+			caps->ac_type = AUDIO_TYPE_FEATURE;
+			caps->ac_format.hw = AUDIO_FU_INP_GAIN;
+			ret = ioctl(fd, AUDIOIOC_GETCAPS, (unsigned long)caps);
+			if (ret < 0) {
+				auddbg("Fail to ioctl AUDIOIOC_GETCAPS, ret = %d\n", ret);
+				return AUDIO_MANAGER_DEVICE_FAIL;
```

Comment:
same as above, additionally, file still opened

Reply:
So, goto error_with_fd & error_out will be helpful.

Reply:
I just added 'close(fd)' since the goto error logic is somewhat burden to me..